scaler/main.go

```diff
@@ -62,8 +62,27 @@ func main() {
 		os.Exit(1)
 	}
 
+	// This callback function is used to fetch and save
+	// the current queue counts from the interceptor immediately
+	// after updating the routingTable information.
+	callbackWhenRoutingTableUpdate := func() error {
+		if err := pinger.fetchAndSaveCounts(ctx); err != nil {
+			return err
+		}
+		return nil
+	}
```

Comment:
@tpiperatgod instead of fetching and saving counts, what do you think about simply adding the hostname to the scaler's in-memory map and letting the pinger fetch counts as it currently does in its own loop?

Reply:
Can I understand your suggestion in this way: 
1. pass in the **pinger** to `StartConfigMapRoutingTableUpdater`
2. then add the element with key the **hostname** to `pinger.allCounts` in the processing

If so, then I used a generic function `cbFunc` instead of passing in **pinger** directly, considering that **pinger** only runs in the scaler service, while `StartConfigMapRoutingTableUpdater` is used by both the scaler and the interceptor.

As well `fetchAndSaveCounts` is an already existing method, and it doesn't need to pass additional arguments, which helps make `callbackWhenRoutingTableUpdate` a generic function, so maybe I should think it makes sense in this scenario?

Please correct me if I have misunderstood...

Reply:
At the same time, the pinger will keep its loop.

Reply:
I think it is also workable to pass in **pinger**, and I will give priority to your suggestion.

Reply:
>Can I understand your suggestion in this way:


@tpiperatgod that would be one way to do it. it looks like a larger refactor might be in order, which we can certainly do in a follow-up pull request. if this works now as-is, then I'm fine with it!

for the follow-up, one possible refactor might be to have `StartConfigMapRoutingTableUpdater` pass more information to the callback about what was updated, so that we can then immediately reflect the updates in the pinger. for example, we could pass the `watch.Event` to the callback. I think we can look into this in more depth after this is merged.